x-pack/plugins/alerting/server/types.ts

```diff
@@ -89,7 +89,7 @@ export interface RuleExecutorOptions<
   InstanceContext extends AlertInstanceContext = never,
   ActionGroupIds extends string = never
 > {
-  alertId: string;
+  alertId: string; // Why is this still called `alertId`? It is actually the rule id
```

Comment:
@CoenWarmer This is piece of technical debt from the framework (issue to address here: https://github.com/elastic/kibana/issues/100115). `alertId` is a legacy naming convention and you are correct it is the rule id.

Reply:
Thanks for clearing this up for me @ymao1! Will update code comment accordingly.